tsfc/driver.py

```diff
@@ -42,6 +42,7 @@ def compile_form(form, prefix="form", parameters=None):
                            do_apply_integral_scaling=True,
                            do_apply_geometry_lowering=True,
                            do_apply_restrictions=True,
+                           preserve_geometry_types=(CellVolume, FacetArea),
```

Comment:
On affine cells, UFL lowers `CellVolume` and `FacetArea` to reference quantities by default, while leaving `CellVolume` and `FacetArea` nodes intact on non-affine cells. This line leaves the handling of these nodes for TSFC even for affine cells. The TSFC pipeline seems to be more robust.